### PR TITLE
ETFE-4187 Audit submission

### PR DESCRIPTION
**Pull request by grahampaulcook**

### CODE REVIEW

File: app/models/audit/SubmitChangeDestinationAudit.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          "status" -> "success",
+          "receipt" -> success.receipt,
+          "receiptDate" -> receiptDate,
+          "responseCode" -> OK
+        )
+      case Left(failedMessage) =>
+        Json.obj(fields =
+          "status" -> "failed",
+          "failedMessage" -> failedMessage.message
+        )
```
Comment (james-work-account) [inline app/models/audit/SubmitChangeDestinationAudit.scala:48]:
```suggestion
        ).deepMerge(
          failedMessage.statusCode.fold(Json.obj())(statusCode => Json.obj("responseCode" -> statusCode))
        )
```
from looking through CaM, it looks like there's a `deepMerge` missing here

File: conf/application.conf [hunk 2]
```diff
@@ -173,4 +173,8 @@ constants {
 
 beta {
   serviceName = "changeDestination"
+}
+
+auditing {
+  enabled = true
```
Comment (james-work-account) [inline conf/application.conf:179]: Other frontends either don't have this set (it defaults to `false` through Bootstrap), or it's set to `false` manually locally - was this left `true` by mistake?

**Review by james-work-account — CHANGES_REQUESTED** [review]

File: conf/application.conf (hunk 2 above)
Comment (grahampaulcook) [reply conf/application.conf]: Yes, that was a mistake, fixed on https://github.com/hmrc/emcs-tfe-change-destination-frontend/pull/53/commits/8ca7e26997e221c5f801c318122711ae9316c355

File: app/models/audit/SubmitChangeDestinationAudit.scala (hunk 1 above)
Comment (grahampaulcook) [reply app/models/audit/SubmitChangeDestinationAudit.scala:48]: deepMerge is only present for CAM, none of the other frontend's have it so I followed suit with those.

**Review by james-work-account — APPROVED** [review]

**Pull request merged**